### PR TITLE
Issue #34

### PR DESCRIPTION
**Pull request by GitPascalP**
report on implementations and further discussion points in #34

### CODE REVIEW

File: examples/userdefined_constraints.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                                 super_episode_length=(8000, 12000))
+
+# constraints, as gym.space (when bounds aren't symmetrical) or np.ndarray
+constraints_1 = Box(low=-0.9, high=0.9, shape=(5,))
+constraints_2 = Box(low=np.array([-70, -440, -70, -300, -420]),
+                    high=np.array([70, 440, 70, 300, 420]))
+# for 70% of normal limits
+discount_factor = 0.7
+
+ConMon = ConstraintMonitor(constraints=constraints_2)
```
Comment (atra94) [inline examples/userdefined_constraints.py:36]: please write instance variables always in snake_case notation (cf. PEP8).

File: gym_electric_motor/core.py [hunk 2]
```diff
@@ -550,6 +550,7 @@ def _check_limit_violation(self, state):
         Returns:
             bool: True, if any observed limit has been violated, False otherwise.
         """
+        # can be overwriten by child-class
```
Comment (atra94) [inline gym_electric_motor/core.py:553]: That's obvious. I think this comment can be deleted.

File: gym_electric_motor/reward_functions/constraint_monitor.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+# todo docstrings, comments for discount factor
+
+
+class ConstraintMonitor:
+    """
+    The ConstraintMonitor Class monitors the system-states and assesses whether
+    they comply the given limits or violate them.
+    It returns the necessary information for the RewardFunction, to calculate
+    the corresponding reward-value.
+    Limits, here called constraints, can be given by the physical limits from
```
Comment (atra94) [inline gym_electric_motor/reward_functions/constraint_monitor.py:15]: The idea behind constraints is that they more or less generalize limits as already mentioned in #34 .

File: gym_electric_motor/core.py [hunk 4]
```diff
@@ -450,6 +450,7 @@ def __init__(self, observed_states='currents', **__):
         self._observed_states = observed_states
         self._reference_generator = None
         self._limits = None
+  #      self._constraint_monitor = None
```
Comment (atra94) [inline gym_electric_motor/core.py:453]: Delete this line, please

**Comment by GitPascalP** [comment]
code is cleaned up and merged with the newest nightly.

**Pull request merged**